includes/REST_API/Stories_Base_Controller.php

```diff
@@ -81,7 +82,7 @@ protected function prepare_item_for_database( $request ) {
 		}
 
 		if ( isset( $request['content'] ) ) {
-			$prepared_post->post_content = $this->base64_decode( $prepared_post->post_content );
+			$prepared_post->post_content = ( new Decoder( new Experiments() ) )->base64_decode( $prepared_post->post_content );
```

Comment:
Can we pass Experiments instanance into the structure like the other classes?

Reply:
Not easily possible, because we don't instantiate the class. WordPress does so in `\WP_Post_Type::get_rest_controller()`

That means in `Stories_Controller` we need to do:

```php
public function __construct( $post_type ) {
	parent::__construct( $post_type, new Experiments() );
}
```

And in `Stories_Base_Controller` we need to do:

```php
public function __construct( $post_type, Experiments $experiments ) {
	parent::__construct( $post_type );
	$this->experiments = $experiments;
}
```

And because the template post type uses `Stories_Base_Controller` directly for its REST controller, we need a new `Templates_Controller` class that simply does:

```php
class Templates_Controller extends Stories_Base_Controller {
	public function __construct( $post_type, Experiments $experiments ) {
		parent::__construct( $post_type );
		$this->experiments = $experiments;
	}
}
```

Reply:
Done in d11e2c79f779e2453dde78a70bb035123b47918f

Reply:
To answer my own question, no. The class is setup here. https://github.com/WordPress/wordpress-develop/blob/f36c42c70034348c4007157c1fc8a8d69e50c90c/src/wp-includes/class-wp-post-type.php#L708-L732

Reply:
Because we use a base class it _does_ work.